go/vt/wrangler/reparent.go

```diff
@@ -757,47 +758,75 @@ func (wr *Wrangler) findCurrentMaster(tabletMap map[string]*topo.TabletInfo) *to
+	alias    *topodatapb.TabletAlias
+	position mysql.Position
+}
+
+func (t *tabletReplInfoAggregator) processTablet(tablet *topodatapb.Tablet) {
+	defer t.waitGroup.Done()
+	t.wrangler.logger.Infof("getting replication position from %v", topoproto.TabletAliasString(tablet.Alias))
 
-	slaveStatusCtx, cancelSlaveStatus := context.WithTimeout(maxPosSearch.ctx, maxPosSearch.waitReplicasTimeout)
+	slaveStatusCtx, cancelSlaveStatus := context.WithTimeout(t.ctx, t.waitReplicasTimeout)
 	defer cancelSlaveStatus()
 
-	status, err := maxPosSearch.wrangler.tmc.SlaveStatus(slaveStatusCtx, tablet)
+	status, err := t.wrangler.tmc.SlaveStatus(slaveStatusCtx, tablet)
 	if err != nil {
-		maxPosSearch.wrangler.logger.Warningf("failed to get replication status from %v, ignoring tablet: %v", topoproto.TabletAliasString(tablet.Alias), err)
+		t.wrangler.logger.Warningf("failed to get replication status from %v, ignoring tablet: %v", topoproto.TabletAliasString(tablet.Alias), err)
 		return
 	}
-	replPos, err := mysql.DecodePosition(status.Position)
+	replPos, err := decodePosition(status)
 	if err != nil {
-		maxPosSearch.wrangler.logger.Warningf("cannot decode slave %v position %v: %v", topoproto.TabletAliasString(tablet.Alias), status.Position, err)
+		t.wrangler.logger.Warningf("cannot decode slave %v position %v: %v", topoproto.TabletAliasString(tablet.Alias), status.Position, err)
 		return
 	}
 
-	maxPosSearch.maxPosLock.Lock()
-	if maxPosSearch.maxPosTablet == nil || !maxPosSearch.maxPos.AtLeast(replPos) {
-		maxPosSearch.maxPos = replPos
-		maxPosSearch.maxPosTablet = tablet
+	t.statusLock.Lock()
+	t.tabletReplInfos = append(t.tabletReplInfos, &replicationInfo{
+		alias:    tablet.Alias,
+		position: replPos,
+	})
+	t.statusLock.Unlock()
+}
+
+// unpackTabletAliases will turn a []*replicationInfo from the tabletReplInfos field into an []*TabletAlias.
+func (t *tabletReplInfoAggregator) unpackTabletAliases() []*topodatapb.TabletAlias {
+	t.statusLock.Lock()
+	tabletAliases := make([]*topodatapb.TabletAlias, 0, len(t.tabletReplInfos))
+
+	for i := range t.tabletReplInfos {
+		tabletAliases = append(tabletAliases, t.tabletReplInfos[i].alias)
 	}
-	maxPosSearch.maxPosLock.Unlock()
+	t.statusLock.Unlock()
+
+	return tabletAliases
+}
+
+func (t *tabletReplInfoAggregator) Len() int {
+	return len(t.tabletReplInfos)
+}
+func (t *tabletReplInfoAggregator) Swap(i, j int) {
+	t.tabletReplInfos[i], t.tabletReplInfos[j] = t.tabletReplInfos[j], t.tabletReplInfos[i]
+}
+func (t *tabletReplInfoAggregator) Less(i, j int) bool {
+	return !t.tabletReplInfos[i].position.AtLeast(t.tabletReplInfos[j].position)
 }
 
 // chooseNewMaster finds a tablet that is going to become master after reparent. The criteria
-// for the new master-elect are (preferably) to be in the same cell as the current master, and
-// to be different from avoidMasterTabletAlias. The tablet with the largest replication
+// for the new master-elect are to be in the same cell as the current master, and
```

Comment:
@enisoc do you know why we limit the search to master's current cell?

Reply:
It makes sense that we'd want to prefer the same cell when possible, as the comment originally said, but I don't know if it was intentional that this code has actually been requiring the same cell. It does make sense from a safety perspective, but I was also surprised by this since the vtctl command docs don't mention it.

In any case, I advised @PrismaPhonic that we should probably preserve the behavior rather than fixing it to match the comment, since users may have come to rely on this and unexpectedly reparenting to a remote cell could cause an outage if the app isn't prepared for it. What do you think?

On a related note, PRS used to only contact tablets in the same cell to get their positions, but now we will contact tablets in all cells and throw away most of them. Do you think it's worth fixing this to avoid the unnecessary calls?